### PR TITLE
Improve mysql user experience

### PR DESCRIPTION
**Pull request by msJinLei**
* Provide hint in doc to use Update-AzMySqlServer &
Update-AzMySqlServerConfiguration
  - https://github.com/Azure/azure-powershell/issues/11954
* Change New-AzMySqlServerReplica name and “replica” instead of name
  - https://github.com/Azure/azure-powershell/issues/11939
  - https://github.com/Azure/azure-powershell/issues/11938
* *-AzMySqlFirewallRule Use only one parameter when only one IP
needs to be authorized
  - https://github.com/Azure/azure-powershell/issues/11933

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/MySql/custom/New-AzMySqlFirewallRule.ps1 [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Creates a new firewall rule or updates an existing firewall rule.
+.Description
+Creates a new firewall rule or updates an existing firewall rule.
+
+.Link
+https://docs.microsoft.com/en-us/powershell/module/az.mysql/new-azmysqlfirewallrule
```
Comment (mikefrobbins) [inline src/MySql/custom/New-AzMySqlFirewallRule.ps1:23]: URL's should not include culture.
```suggestion
https://docs.microsoft.com/powershell/module/az.mysql/new-azmysqlfirewallrule
```

File: src/MySql/custom/Update-AzMySqlConfiguration.ps1 [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+<#
+.Synopsis
+Updates a configuration of a server.
+Use Update-AzMySqlServer instead if you want update AdministratorLoginPassword, sku, etc.
+.Description
+Updates a configuration of a server.
+Use Update-AzMySqlServer instead if you want update AdministratorLoginPassword, sku, etc.
+
+.Link
+https://docs.microsoft.com/en-us/powershell/module/az.mysql/update-azmysqlconfiguration
```
Comment (mikefrobbins) [inline src/MySql/custom/Update-AzMySqlConfiguration.ps1:25]:
```suggestion
https://docs.microsoft.com/powershell/module/az.mysql/update-azmysqlconfiguration
```

File: src/MySql/custom/Update-AzMySqlFirewallRule.ps1 [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+# ----------------------------------------------------------------------------------
+
+<#
+.Synopsis
+Creates a new firewall rule or updates an existing firewall rule.
+.Description
+Creates a new firewall rule or updates an existing firewall rule.
+
+.Link
+https://docs.microsoft.com/en-us/powershell/module/az.mysql/update-azmysqlfirewallrule
```
Comment (mikefrobbins) [inline src/MySql/custom/Update-AzMySqlFirewallRule.ps1:23]:
```suggestion
https://docs.microsoft.com/powershell/module/az.mysql/update-azmysqlfirewallrule
```

File: src/MySql/custom/Update-AzMySqlServer.ps1 [hunk 4]
```diff
@@ -16,7 +16,7 @@
 function Update-AzMySqlServer {
     [OutputType([Microsoft.Azure.PowerShell.Cmdlets.MySql.Models.Api20171201.IServer])]
     [CmdletBinding(DefaultParameterSetName='UpdateExpanded', PositionalBinding=$false, SupportsShouldProcess, ConfirmImpact='Medium')]
-    [Microsoft.Azure.PowerShell.Cmdlets.MySql.Description('Updates an existing server. The request body can contain one to many of the properties present in the normal server definition.')]
+    [Microsoft.Azure.PowerShell.Cmdlets.MySql.Description('Updates an existing server. Use Update-AzMySqlConfiguration instead if you want update server parameters such as wait_timeout, net_retry_count. The request body can contain one to many of the properties present in the normal server definition.')]
```
Comment (mikefrobbins) [inline src/MySql/custom/Update-AzMySqlServer.ps1:19]:
```suggestion
    [Microsoft.Azure.PowerShell.Cmdlets.MySql.Description('Updates an existing server. The request body can contain one to many of the properties present in the normal server definition. Use Update-AzMySqlConfiguration instead if you want update server parameters such as wait_timeout or net_retry_count.')]
```

File: src/MySql/docs/Az.MySql.md [hunk 5]
```diff
@@ -58,12 +58,14 @@ Restore a server from an existing backup
...
 Updates a configuration of a server.
+Use Update-AzMySqlServer instead if you want update AdministratorLoginPassword, sku, etc.
 
 ### [Update-AzMySqlFirewallRule](Update-AzMySqlFirewallRule.md)
 Creates a new firewall rule or updates an existing firewall rule.
 
 ### [Update-AzMySqlServer](Update-AzMySqlServer.md)
 Updates an existing server.
+Use Update-AzMySqlConfiguration instead if you want update server parameters such as wait_timeout, net_retry_count.
 The request body can contain one to many of the properties present in the normal server definition.
```
Comment (mikefrobbins) [inline src/MySql/docs/Az.MySql.md:69]:
```suggestion
Updates an existing server.
The request body can contain one to many of the properties present in the normal server definition.
Use Update-AzMySqlConfiguration instead if you want update server parameters such as wait_timeout or net_retry_count.
```

File: src/MySql/docs/Update-AzMySqlConfiguration.md [hunk 6]
```diff
@@ -9,6 +9,7 @@ schema: 2.0.0
 
 ## SYNOPSIS
 Updates a configuration of a server.
+Use Update-AzMySqlServer instead if you want update AdministratorLoginPassword, sku, etc.
```
Comment (mikefrobbins) [inline src/MySql/docs/Update-AzMySqlConfiguration.md:12]: Having this information in the description is sufficient. Having it in both the synopsis and description seems redundant.
```suggestion
Updates a configuration of a server.
```

File: src/MySql/docs/Update-AzMySqlServer.md [hunk 7]
```diff
@@ -9,6 +9,7 @@ schema: 2.0.0
 
 ## SYNOPSIS
 Updates an existing server.
+Use Update-AzMySqlConfiguration instead if you want update server parameters such as wait_timeout, net_retry_count.
 The request body can contain one to many of the properties present in the normal server definition.
```
Comment (mikefrobbins) [inline src/MySql/docs/Update-AzMySqlServer.md:13]:
```suggestion
Updates an existing server.
The request body can contain one to many of the properties present in the normal server definition.
```

File: src/MySql/docs/Update-AzMySqlServer.md [hunk 8]
```diff
@@ -33,6 +34,7 @@ Update-AzMySqlServer -InputObject <IMySqlIdentity> [-AdministratorLoginPassword
 
 ## DESCRIPTION
 Updates an existing server.
+Use Update-AzMySqlConfiguration instead if you want update server parameters such as wait_timeout, net_retry_count.
 The request body can contain one to many of the properties present in the normal server definition.
```
Comment (mikefrobbins) [inline src/MySql/docs/Update-AzMySqlServer.md:38]:
```suggestion
Updates an existing server.
The request body can contain one to many of the properties present in the normal server definition.
Use Update-AzMySqlConfiguration instead if you want update server parameters such as wait_timeout or net_retry_count.
```

File: src/MySql/docs/Update-AzMySqlConfiguration.md (hunk 6 above)
Comment (msJinLei) [reply src/MySql/docs/Update-AzMySqlConfiguration.md:12]: @mikefrobbins  we need the line of "Use Update-AzMySqlServer instead if you want update AdministratorLoginPassword, sku, etc.". If the statement is not good enough, could you provide some proposals to improve it? Thanks

File: src/MySql/custom/New-AzMySqlFirewallRule.ps1 (hunk 1 above)
Comment (msJinLei) [reply src/MySql/custom/New-AzMySqlFirewallRule.ps1]: Link should be auto generated. It is added by mistake.  We will fix it in powershell generator.

File: src/MySql/custom/Update-AzMySqlConfiguration.ps1 (hunk 2 above)
Comment (msJinLei) [reply src/MySql/custom/Update-AzMySqlConfiguration.ps1]: Link should be auto generated. It is added by mistake.  We will fix it in powershell generator.

File: src/MySql/custom/Update-AzMySqlFirewallRule.ps1 (hunk 3 above)
Comment (msJinLei) [reply src/MySql/custom/Update-AzMySqlFirewallRule.ps1]: Link should be auto generated. It is added by mistake.  We will fix it in powershell generator.

**Pull request merged**